File: c#/p1.cs
public class P1
{
  public static void Main()
  {
    int sum = 0;
    for(int i = 0; i < 1000; i++)
    {
      if(i % 3 == 0 || i % 5 == 0)
      {
        sum += i;
      }
    }
    System.Console.WriteLine(sum);
  }
}

File: rust/p1.rs
fn main() {
  println!("{}", p1(0, 0));
}

fn p1(i: u32, sum: u32) -> u32 {
  if i == 1000 {
    sum
  } else {
    if i % 5 == 0 || i % 3 == 0 {
      p1(i + 1, sum + i)
    } else{
      p1(i + 1, sum)
    }
  }
}